pkg/build/generator/generator.go

```diff
@@ -702,3 +710,18 @@ func getStrategyImageChangeTrigger(bc *buildapi.BuildConfig) *buildapi.ImageChan
 	}
 	return nil
 }
+
+// getImageChangeTriggerForRef returns the ImageChangeTrigger that is triggered by a change to
+// the provided object reference, if any
+func getImageChangeTriggerForRef(bc *buildapi.BuildConfig, ref *kapi.ObjectReference) *buildapi.ImageChangeTrigger {
+	if ref == nil || ref.Kind != "ImageStreamTag" {
+		return nil
+	}
+	for _, trigger := range bc.Spec.Triggers {
+		if trigger.Type == buildapi.ImageChangeBuildTriggerType && trigger.ImageChange.From != nil &&
```

Comment:
if i understand your scenario:

you have a strategy.From that matches an ImageSource.From
you have a default ICT

you're concerned that while the strategy will pick up the imageid from the trigger, the imagesource will get its imageid from a call to resolve?

yes i guess that could happen, will see what i can do.